### DISCUSSION TITLE
Where is the image PATH set exactly?

### DISCUSSION THREAD

**Discussion by petergerdy** [lllyasviel/Fooocus#2220] [General]
Hi,

I see that the path is set to the user's home path. Where can this be changed and where in the code is the file exactly stored to that path? Is this somewhere in Gradio or in Fooocus? 

Thanks for suggestions :) Loving Fooocus!

Peter

**Comment by mashb1t** [comment]
Please find the value for "path_outputs'" in your config.txt file and change it to your liking.

Be aware that changing this outside of your fooocus root directory will lead to the history log being inaccessible due to security measures. Saving files will still work though.

**Reply by petergerdy** [reply]
I know the path_outputs, its set to ""path_outputs": "D:\\fooocus\\Fooocus\\outputs"" in my config.

However, the images are stored TWICE, once in path_outputs and once in the users home directory "https://www.domain.com/file=C:/Users/XXXXXX~1/AppData/Local/Temp/gradio/2342K3H423JK4H23KJ4/image.png

Can this folder even be defined from within fooocus or is this a Gradio thing?

**Reply by ETCore7** [reply]
Gradio folder under Temp doesn't exist here.

**Reply by adjoth** [reply]
I've beem trying to figure this out as well. I was able to locate, and remove, the images in the output folder as well as the temp folder. But Fooocus is still "seeing" my history (even after restarting the application). I'd be nice if there was a proper means of clearning your history or making it private or a setting to allow periodic clean-ups. I have a few ideas how this may be done but I suspect the model maybe storing them elsewhere which is making life difficult.

**Comment by mashb1t** [comment]
this is a gradio setting, but can be overridden using an env var. PR https://github.com/lllyasviel/Fooocus/pull/1992
